EventListener/TagSubscriber.php

```diff
@@ -86,8 +113,13 @@ public function onKernelResponse(FilterResponseEvent $event)
         }
 
         if ($request->isMethodSafe()) {
-            // For safe requests (GET and HEAD), set cache tags on response
-            $this->cacheManager->tagResponse($response, $tags);
+            if ($event->isMasterRequest()) {
+                // For safe requests (GET and HEAD), set cache tags on response
+                $this->cacheManager->tagResponse($response, $tags);
+            } else {
+                // Retain tags to attach to response to master request
+                $this->tags = $tags;
+            }
```

Comment:
@stof does this make sense to you or am i missing something? it seems to me we only want to tag on the response to master request.

the current implementation introduces an inconsistency in that tags from configuration / annotation are preserved from subrequest controllers, while cache control of subrequests is lost. but i think that is ok, as such information set on a response object is lost in twig subrequests in general.